airflow-provider-fivetran/operators/fivetran.py

```diff
@@ -48,121 +52,24 @@ def __init__(
-        log.info(
-            "Connector type: {}, connector schema: {}".format(
-                connector_details["service"], connector_details["schema"]
-            )
-        )
-        log.info("Connectors logs at {}".format(URL_LOGS))
-
-        # Set connector to manual sync mode, required to force sync through the API
-        resp = session.patch(
-            URL_CONNECTOR,
-            data=json.dumps({"schedule_type": "manual"}),
-            headers={"Content-Type": "application/json;version=2"},
-            auth=(self.api_key, self.api_secret),
-        )
-        # Start connector sync
-        resp = session.post(
-            URL_CONNECTOR + "/force", auth=(self.api_key, self.api_secret)
-        )
-
-        loop: bool = True
-        while loop:
-            resp = session.get(URL_CONNECTOR, auth=(self.api_key, self.api_secret))
-            current_details = resp.json()["data"]
-            # Failsafe, in case we missed a state transition – it is possible with a long enough
-            # `poll_status_every_n_seconds` we could completely miss the 'syncing' state
-            succeeded_at = self.parse_timestamp(current_details["succeeded_at"])
-            failed_at = self.parse_timestamp(current_details["failed_at"])
-            current_completed_at = (
-                succeeded_at if succeeded_at > failed_at else failed_at
-            )
-            # The only way to tell if a sync failed is to check if its latest failed_at value
-            # is greater than then last known "sync completed at" value.
-            if failed_at > previous_completed_at:
-                raise ValueError(
-                    'Fivetran sync for connector "{}" failed; please see logs at {}'.format(
-                        self.connector_id, URL_LOGS
-                    )
-                )
-            # Started sync will spend some time in the 'scheduled' state before
-            # transitioning to 'syncing'.
-            # Capture the transition from 'scheduled' to 'syncing' or 'rescheduled',
-            # and then back to 'scheduled' on completion.
-            sync_state = current_details["status"]["sync_state"]
-            log.info(
-                'Connector "{}" current sync_state = {}'.format(
-                    self.connector_id, sync_state
-                )
-            )
-            if current_completed_at > previous_completed_at:
-                loop = False
-            else:
-                time.sleep(self.poll_status_every_n_seconds)
-
-        return {
-            "succeeded_at": succeeded_at.to_iso8601_string(),
-            "connector_id": self.connector_id,
-        }
+        hook = self._get_hook()
+        resp = hook.check_connector(self.connector_id)
+        resp = hook.set_manual_schedule(self.connector_id)
+        resp = hook.start_fivetran_sync(self.connector_id)
+        resp = hook.start_fivetran_sync(self.connector_id, self.poll_status_every_n_seconds)
```

Comment:
We are going to want to break this out into different methods.

Reply:
+1 on breaking this out into different methods

As you noted in the `poll_fivetran_sync` function's to-do, this function is a good candidate for a sensor. The [HTTP sensor](https://github.com/apache/airflow/blob/master/airflow/providers/http/sensors/http.py#L99) provides a clear, related example.

``` python
def poke(self, context: Dict[Any, Any]) -> bool:
        from airflow.utils.operator_helpers import make_kwargs_callable

        self.log.info('Poking: %s', self.endpoint)
        try:
            response = self.hook.run(
                self.endpoint,
                data=self.request_params,
                headers=self.headers,
                extra_options=self.extra_options,
            )
            if self.response_check:
                kwargs_callable = make_kwargs_callable(self.response_check)
                return kwargs_callable(response, **context)

        except AirflowException as exc:
            if str(exc).startswith("404"):
                return False

            raise exc

        return True
```